crates/core/src/token.rs

```diff
@@ -231,13 +231,14 @@ impl Amount {
         Self { raw: Uint(raw) }
     }
 
-    /// Given a u128 and [`MaspDigitPos`], construct the corresponding
+    /// Given a i128 and [`MaspDigitPos`], construct the corresponding
     /// amount.
-    pub fn from_masp_denominated_u128(
-        val: u128,
+    pub fn from_masp_denominated_i128(
+        val: i128,
         denom: MaspDigitPos,
     ) -> Option<Self> {
         let lo = u64::try_from(val).ok()?;
```

Comment:
ah, I think this fn is not quite right - I think I messed it up in #3074 and now it will return `None` for anything greater than `u64::MAX` due to this conversion. Previously, it was doing a truncating cast (`let lo = val as u64;` so I think we need do something like that again

Reply:
Adjusted in 6026c4217ec5d87bdc903b09b5812a36ad58d2ea